src/metabase/driver/postgres.clj

```diff
@@ -209,6 +213,94 @@
 
 (def ^:private ^:dynamic *enum-types* nil)
 
+(def ^:private table-type-clauses
+  {"TABLE"              [:raw "c.relkind = 'r' AND n.nspname !~ '^pg_' AND n.nspname <> 'information_schema'"]
+   "PARTITIONED TABLE"  [:raw "c.relkind = 'p' AND n.nspname !~ '^pg_' AND n.nspname <> 'information_schema'"]
+   "VIEW"               [:raw "c.relkind = 'v' AND n.nspname <> 'pg_catalog' AND n.nspname <> 'information_schema'"]
+   "FOREIGN TABLE"      [:raw "c.relkind = 'f'"]
+   "MATERIALIZED VIEW"  [:raw "c.relkind = 'm'"]})
+
+(defn- get-tables-sql
+  [schemas tables types]
+  ;; Should track this implementation https://github.com/davecramer/pgjdbc/blob/a714bfd/pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java#L1272
+  (sql/format
+   (cond-> {:select    [[:n.nspname :schema]
+                        [:c.relname :name]
+                        [[:case-expr [:or [(keyword "~") :n.nspname "^pg_"] [:= :n.nspname "information_schema"]]
+                          true ;; system tables
+                          [:case
+                           [:or [:= :n.nspname "pg_catalog"] [:= :n.nspname "information_schema"]]
+                           [:case-expr :c.relkind
+                            [:inline "r"] [:inline "SYSTEM TABLE"]
+                            [:inline "v"] [:inline "SYSTEM VIEW"]
+                            [:inline "i"] [:inline "SYSTEM INDEX"]
+                            :else nil]
+                           [:= :n.nspname [:inline "pg_toast"]]
+                           [:case-expr :c.relkind
+                            [:inline "r"] [:inline "SYSTEM TOAST TABLE"]
+                            [:inline "i"] [:inline "SYSTEM TOAST INDEX"]
+                            :else nil]
+
+                           [:case-expr :c.relkind
+                            [:inline "r"] [:inline "TEMPORARY TABLE"]
+                            [:inline "p"] [:inline "TEMPORARY TABLE"]
+                            [:inline "i"] [:inline "TEMPORARY INDEX"]
+                            [:inline "s"] [:inline "TEMPORARY SEQUENCE"]
+                            [:inline "v"] [:inline "TEMPORARY VIEW"]
+                            :else nil]]
```

Comment:
We don't care about all these table types, so we should just filter them out instead of making this case expression longer.